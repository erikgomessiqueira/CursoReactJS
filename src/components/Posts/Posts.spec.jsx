import { render, screen } from '@testing-library/react';
import { Posts } from '.';

const props = {
  posts: [
    {
      id: 1,
      title: 'title01',
      body: 'body01',
      cover: 'img/img01.png',
    },
    {
      id: 2,
      title: 'title02',
      body: 'body02',
      cover: 'img/img02.png',
    },
    {
      id: 3,
      title: 'title03',
      body: 'body03',
      cover: 'img/img03.png',
    },
  ],
};

describe('<Posts/>', () => {
  it('should render posts', () => {
    render(<Posts {...props} />);

    const allTitles = screen.getAllByRole('heading', { name: /title/i });
    expect(allTitles).toHaveLength(3);

    const allImages = screen.getAllByRole('img', { name: /title/i });
    expect(allImages).toHaveLength(3);

    const allBody = screen.getAllByText(/body/i);
    expect(allBody).toHaveLength(3);

    const oneImage = screen.getByRole('img', { name: /title02/i });
    expect(oneImage).toHaveAttribute('src', 'img/img02.png');
  });

  it('should not render posts', () => {
    render(<Posts />);

    const postEmpty = screen.queryByRole('heading', { name: /title/i });
    expect(postEmpty).not.toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Posts {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
