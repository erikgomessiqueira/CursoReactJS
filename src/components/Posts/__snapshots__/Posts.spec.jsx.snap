// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts/> should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="title01"
      src="img/img01.png"
    />
    <div
      class="post-content"
    >
      <h1>
        title01
         
        1
      </h1>
      <p>
        body01
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title02"
      src="img/img02.png"
    />
    <div
      class="post-content"
    >
      <h1>
        title02
         
        2
      </h1>
      <p>
        body02
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="title03"
      src="img/img03.png"
    />
    <div
      class="post-content"
    >
      <h1>
        title03
         
        3
      </h1>
      <p>
        body03
      </p>
    </div>
  </div>
</div>
`;
